include/valijson/validation_visitor.hpp

```diff
@@ -839,9 +844,13 @@ class ValidationVisitor: public constraints::ConstraintVisitor
             return true;
         }
 
-        const std::regex patternRegex(constraint.getPattern<std::string::allocator_type>());
+        std::string pattern(constraint.getPattern<std::string::allocator_type>());
+        auto it = m_regexesCache.find(pattern);
```

Comment:
Thinking about this from a performance perspective, I wonder if you might see some more improvement by adding a function to the PatternConstraint class to access the string's c_str value directly (e.g. `getPatternCStr()`). Then you could defer copying the string until it is actually emplaced into the map.

Reply:
I don't see any improvement for my use case. May be for higher amount of validations that could be significant.
Funny, but std::ref gave me more than passing raw char pointer.

Reply:
It can be hard to predict exactly how different approaches will work these days. Seeing as this has already improved performance for a real use case, I'm happy to merge it as is.